src/Access/AccessRights.cpp

```diff
@@ -284,77 +341,100 @@ struct AccessRights::Node
         node_name = src.node_name;
         level = src.level;
         flags = src.flags;
+        wildcard_grant = src.wildcard_grant;
         min_flags_with_children = src.min_flags_with_children;
         max_flags_with_children = src.max_flags_with_children;
         if (src.children)
-            children = std::make_unique<std::unordered_map<std::string_view, Node>>(*src.children);
+            children = std::make_unique<Container>(*src.children);
         else
             children = nullptr;
         return *this;
     }
 
+    bool isLeaf() const { return node_name.empty(); }
```

Comment:
The name `isLeaf()` is confusing. I mean the true `isLeaf()` function should be like this
```
bool isLeaf() const { return children.empty(); }
```

Reply:
In our scenario, we can have subtrees on the leaves, which is why the naming may seem confusing.